### PR TITLE
add slovak language localization

### PR DESCRIPTION
**Pull request by filiphric**
Hey there!

Awesome work on the watch face, I love it 😍  

I tried putting together a Slovak localization. I’m not really fluent in the Monkey C language, so I had to find my way around the code a little. I added couple of time override categories as some of the times in Slovak language require that.

Hope you’ll like it, let me know what you thing, I’ll be super happy if this gets released.

Have a great day! 👋

### CODE REVIEW

**Comment by filiphric** [comment]
Also, I was wondering if there is a way of adding non-breaking space, without breaking the calculation of width 🤔

**Comment by nels0nwu** [comment]
Wow I'm impressed you've figured so much out, even the fonts. I'll have to check it out!

I've used `|` in some translations to limit the number of rows. Is that what your after?

**Comment by filiphric** [comment]
Thanks, the code was very readable IMO, you’ve done a great job there 🙂
I used bmfont for the fonts, I think you used the same? I had to manually change xadvance parameter for character 357, because bmfont partially clips it. I couldn’t find a way for setting it up, so this is something that is missing in the `resources/fonts/roboto_text.bmfc` setting file. 

I completely missed the `|` for non-breaking spaces, l’ll test it out and update the PR.

**Comment by filiphric** [comment]
Hmm ok it seems I broke other languages by adding those additional categories like `textHalfHourOverridesArray` and
`textQuarterHourOverridesArray`. Not sure how to fix that, but I guess they need to be defined in all languages? 🤔

Also, I seem to have removed character 32, which is pretty important 😂 Found out when I played with the `|` symbol. Let me fix that and update the PR.

Any help or pointers with the override categories problems appreciated 🙏

**Comment by filiphric** [comment]
The character 32 is back, trying to fix the thing with new categories

**Comment by filiphric** [comment]
Oh figured it out. Defaults on English language were missing, so exceptions were thrown on other languages. Fixed now, I’d say it’s ready now 🙂

**Comment by nels0nwu** [comment]
I used bmfont as well. Tried replicating the clipping issue.. wow that's frustrating. Wish I could find a solution for that.
image
How were you able to figure out the correct `xadvance` for each font size? I have no idea.

Great job on the translation - time in Slovak is really tricky 🤯

File: resources-slo/strings/strings.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<!-- translation credits: Filip Hric -->
+<strings>
+    <string id="Language">slo</string>
+
+    <string id="DateFormatMonthDate">Month-Date eg. Feb 26</string>
+    <string id="DateFormatDayDate">Day-Date eg. Tue 26</string>
+    <string id="MonthDateFormat">$1$ $2$</string>
+    <string id="DayDateFormat">$1$ $2$</string>
```
Comment (nels0nwu) [inline resources-slo/strings/strings.xml]: unneeded when unchanged from English strings

**Comment by filiphric** [comment]
> How were you able to figure out the correct xadvance for each font size? I have no idea.

Trial and error my friend 😂

I added a little doc inside the folder for future changes. I know it’s not ideal, but if it gets broken, I’d be the first one to notice 🙂. I even tried the solution [mentioned in docs](http://www.angelcode.com/products/bmfont/doc/known_issues.html), but had no luck there.

File: source/FuzzyTextRevolutionView.mc [hunk 2]
```diff
@@ -304,6 +352,27 @@ class FuzzyTextRevolutionView extends WatchUi.WatchFace {
...
+        }
+
+        if (displayMinute == 30) {
+            var override = textHalfHourOverridesArray[displayHour%12];
+            if (!override.equals("")) {
+                unsplit = override;
+            }
+        }
+
+        if (displayMinute == 30) {
```
Comment (nels0nwu) [inline source/FuzzyTextRevolutionView.mc]:
```suggestion
        if (displayMinute == 45) {
```

File: resources-slo/strings/strings.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    <string id="TimeMinute0">0 hodín</string>
+    <string id="TimeMinute5">0|aj päť|minút</string>
+    <string id="TimeMinute10">0|aj desať|minút</string>
+    <string id="TimeMinute15">štvrť na|1</string>
+    <string id="TimeMinute20">0 dvadsať</string>
+    <string id="TimeMinute25">0 dvadsať päť</string>
+    <string id="TimeMinute30">0 tridsať</string>
+    <string id="TimeMinute35">0 tridsať päť</string>
+    <string id="TimeMinute40">0 štyridsať</string>
+    <string id="TimeMinute45">tri štvrte na 1</string>
```
Comment (nels0nwu) [inline resources-slo/strings/strings.xml]: I try to limiting the lines to 3 as 4 can display poorly
same with: tri štvrte na jednu

File: resources-slo/strings/strings.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <string id="TimeMinute0">0 hodín</string>
+    <string id="TimeMinute5">0|aj päť|minút</string>
+    <string id="TimeMinute10">0|aj desať|minút</string>
+    <string id="TimeMinute15">štvrť na|1</string>
+    <string id="TimeMinute20">0 dvadsať</string>
+    <string id="TimeMinute25">0 dvadsať päť</string>
+    <string id="TimeMinute30">0 tridsať</string>
+    <string id="TimeMinute35">0 tridsať päť</string>
+    <string id="TimeMinute40">0 štyridsať</string>
+    <string id="TimeMinute45">tri štvrte|na 1</string>
```
Comment (nels0nwu) [inline resources-slo/strings/strings.xml]:
```suggestion
    <string id="TimeMinute45">tri štvrte|na|1</string>
```
The code needs the hour on a separate line, sorry😅😅

File: resources-slo/strings/strings.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    <string id="HalfHour4Override">pol *piatej</string>
+    <string id="HalfHour5Override">pol *šiestej</string>
+    <string id="HalfHour6Override">pol *siedmej</string>
+    <string id="HalfHour7Override">pol *ôsmej</string>
+    <string id="HalfHour8Override">pol *deviatej</string>
+    <string id="HalfHour9Override">pol *desiatej</string>
+    <string id="HalfHour10Override">pol *jedenástej</string>
+    <string id="HalfHour11Override">pol *dvanástej</string>
+
+    <string id="ThreeQuarterHour0Override">tri štvrte|na *jednu</string>
```
Comment (nels0nwu) [inline resources-slo/strings/strings.xml]:
```suggestion
    <string id="ThreeQuarterHour0Override">tri štvrte|na|*jednu</string>
```

**Review by nels0nwu — CHANGES_REQUESTED** [review]
Really happy with the rest. Keen to merge 😁

File: resources-slo/strings/strings.xml (hunk 5 above)
Comment (filiphric) [reply resources-slo/strings/strings.xml]: no problem, I changed it so the separation is correct

**Pull request merged**